Upgrade to te latest version of kubernetes API.

bokysan opened the pull request:
Hi,

this change upgrades the segment to work with the latest version of Kubernetes API (9.0.0).

bokysan committed:
Upgrade to te latest version of kubernetes API.

bokysan committed:
Coloring and additional configuration

These two ideas were taken from: [powerkube](https://github.com/zcmarine/powerkube)
which seems to be unmaintained:
1. Add colouring in cases when the the name of context / namespace
   matches predefined list.
2. Add the possibility to hide certain segments (e.g. cluster or
   namespace).

bokysan committed:
Optimize display

This commit allows you to hide the namespace name if the name is
'default'.

bokysan committed:
Custom coloring for every segment.

Colors for each of the segments (namespace, cluster) have been split.
It's now possible to hide each segment independently.

It's also possible to hide the Kubernetes logo to further reduce the space
used by the segment.

so0k commented:
looks good, do we need to update https://github.com/so0k/powerline-kubernetes/blob/master/requirements.txt ?

and perhaps bump version in https://github.com/so0k/powerline-kubernetes/blob/master/setup.py

EDIT: and thank you very much for your PR

bokysan commented:
You are correct.

I did fix after the PR but have encountered another issue. I'll update the pull request in a few days.

bokysan committed:
requirements.txt updated and a bug fixed.

The kubernetes config file is now loaded on every `__call__`.
Previously, this file was loaded only once which has led to the
fact that kubernetes context was not properly updaded unless powerline
was restarted.

bokysan commented:
This latest fix repairs the issue.

The commit also allows you to specify a "warning" colour of the segment if you are working in a specific namespace / context.

so0k commented on setup.py:
```diff
@@ -1,18 +1,20 @@
     description      = 'A Powerline segment to show Kubernetes context',
+    long_description = (pathlib.Path(__file__).parent / "README.md").read_text(),
+    long_description_content_type="text/markdown",
     version          = '1.0.2',
```
I think we need to bump this also

bokysan replied:
Will do.

so0k commented on setup.py:
```diff
@@ -1,18 +1,20 @@
     version          = '1.0.2',
     keywords         = 'powerline kubernetes context',
     license          = 'MIT',
     author           = 'Vincent De Smet',
```
do you want to be added here?

bokysan replied:
That would be lovely, thank you.

I'll fix both issues, squash the commits and revert back.

bokysan committed:
Bumped version and added authors to README.md.

bokysan commented:
Took a while but both fixes have been made.

Please merge & publish. Thank you.

so0k commented on README.md:
```diff
@@ -23,26 +26,49 @@ The Kubernetes segment uses a couple of custom highlight groups. You'll need to
 ---
 
 Inspired by [powerline-docker](https://github.com/adrianmo/powerline-docker).
+Inspired by [powerkube](https://github.com/zcmarine/powerkube).
```
this is not correct :D - please remove

https://github.com/so0k/powerline-kubernetes/commit/f1805a0d054f6bf076955de4c0c22f7881aeae38
https://github.com/zcmarine/powerkube/commit/a95986a24f2057727f3019f0245c15af5e64be4f

bokysan replied:
I have removed this line from my master branch, although I did get ideas there (e.g. red color for important contexts).

bokysan replied:
(I'm not quite sure why github is not showing my latest commit on this pull request, though)

so0k replied:
ooh, I see what you mean

bokysan committed:
Remove Inspired by line.

so0k commented:
man I just realised I forgot to push your version to pypi - just did 😅
